software/contrib/master_clock.py

```diff
@@ -251,24 +331,28 @@ def computeGcd(self, x, y):
     def lcm(self, li):
         lcm = 1
         for item in li:
-            lcm = lcm*item//max(self.computeGcd(lcm, item), 1)
+            if item != 0 and item != 'r':
+                lcm = lcm*item//max(self.computeGcd(lcm, item), 1)
         return lcm
 
-    ''' Triggered by main. Sends output pulses at required division '''
+    ''' Sends output pulses at required division '''
     def clockTrigger(self):
 
         if self.DEBUG:
-            print('BPM: ' + str(self.bpm) + ' cycle: ' + str(self.timeToSleepMs) + ' PW:' + str(self.pulseWidthMs))
+            print('BPM: ' + str(self.bpm) + ' cycle: ' + str(self.mSBetweenClockCycles) + ' PW:' + str(self.pulseWidthMs))
 
         for idx, output in enumerate(self.outputDivisions):
-            if output != 0:
+            if output != 'r':
```

Comment:
[question] How could output contain `r`? I can't find where that could happen.

Reply:
See line 83: self.clockDivisions.append('r').
I add this as an option for random triggers/clock pulses.

Reply:
Oh ok, I see it , sorry for that.

Another question then, is it ok (by ok I mean performance wise) to have different variable types in a list?  
Isn't it better to have a single type in the list? In which case you could use `-1` to represent that value.

Might be a stupid question but I'm quite new to Python.

Reply:
I wanted to have 'r' so it is more obvious in the UI r=random.
TBH, I am not sure if there is a performance hit or not, but I did not notice any issues when I changed it from 0 to 'r' in the most recent revision.

Reply:
Understood, thanks :)